lib/plugins/aws/package/compile/events/alb/lib/listenerRules.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const BbPromise = require('bluebird');
+
+module.exports = {
+  compileListenerRules() {
+    this.validated.events.forEach((event) => {
+      const listenerRuleLogicalId = this.provider.naming
+        .getAlbListenerRuleLogicalId(event.functionName, event.priority);
+      const targetGroupLogicalId = this.provider.naming
+        .getAlbTargetGroupLogicalId(event.functionName);
+
+      const Conditions = [
+        {
+          Field: 'path-pattern',
+          Values: [event.conditions.path],
+        },
+      ];
+      if (event.conditions.host) {
+        Conditions.push({
+          Field: 'host-header',
+          Values: [event.conditions.host],
+        });
+      }
+
+      Object.assign(this.serverless.service.provider.compiledCloudFormationTemplate.Resources, {
+        [listenerRuleLogicalId]: {
+          Type: 'AWS::ElasticLoadBalancingV2::ListenerRule',
+          Properties: {
+            Actions: [
+              {
+                Type: 'forward',
+                TargetGroupArn: {
+                  Ref: targetGroupLogicalId,
+                },
+              },
+            ],
+            Conditions,
+            ListenerArn: event.listenerArn,
+            Priority: event.priority,
+          },
+        },
+      });
+    });
+
+    return BbPromise.resolve();
```

Comment:
Yes, this is a convention we've introduced when starting the Framework. The main reason being that the Plugin Manager treats every single plugin as an async entity.

Another reason was that some plugins to some async stuff, while other's don't. This introduced a lot of confusion and non-returned promises because some code got copy and pasted (and hence sync and async got mixed).

Reply:
Ok, it may explain why `package:compileEvents` hook is exposed as a promise returning function.
Still here every internal _sync_ method (not exposed to the Plugin Manager in any way) was configured to return a promise, which seems a bit overkill (that also increases noise to signal ratio among framework modules)

Also when we deal with generic situation where given function might be _async_ or _not_.
Instead of forcing _async_ resolution for any case, we may consume result as:

```javascript
new Promise(resolve => resolve(hookCallback()))
```
Then hook callback may be _sync_ if it's a sync, and be _async_ when needed.

I'm a bit worried about forcing _async_ notation everywhere, seems to raise complexity unnecessary.

Reply:
Yes, I see. That makes sense.

Let me turn the internal code to smth. sync and maybe we can open up another issue to audit the codebase and see where we can change the code to be sync (in cases where it's async right now).

Reply:
We may refactor on the go, gradually, when there's a good moment.

With such large codebase I think it might be very difficult (when agreeing on an improvement) ensure every bit is updated with it.

It might be that the only effective way is just to start somewhere, and with time cover the rest.